packages/react-aria-components/docs/ComboBox.mdx

```diff
@@ -920,31 +922,86 @@ interface Character {
 
 function AsyncLoadingExample() {
   let list = useAsyncList<Character>({
-    async load({signal, filterText}) {
+    async load({signal, cursor, filterText}) {
+      if (cursor) {
+        cursor = cursor.replace(/^http:\/\//i, 'https://');
+      }
+
+      // Brief delay so spinner can be seen more easily
+      await new Promise(resolve => setTimeout(resolve, 500));
       let res = await fetch(
-        `https://swapi.py4e.com/api/people/?search=${filterText}`,
+        cursor || `https://swapi.py4e.com/api/people/?search=${filterText}`,
         {signal}
       );
       let json = await res.json();
 
       return {
-        items: json.results
+        items: json.results,
+        cursor: json.next
       };
     }
   });
 
   return (
     <MyComboBox
       label="Star Wars Character Lookup"
-      items={list.items}
       inputValue={list.filterText}
       onInputChange={list.setFilterText}>
-      {(item) => <ListBoxItem id={item.name}>{item.name}</ListBoxItem>}
+      {/*- begin highlight -*/}
```

Comment:
heh... this is a sorta confusing API. I see how it came about, and I don't have a different suggestion at this time since we're combining static items with dynamic
I guess we could show it by adding a "loading" item to the list.items and rendering the ListBoxLoadMoreItem when we get to that item, but then we might have to remember to remove it and move it to the end of the list

Reply:
yeahhh, it certainly is. If definitely would be nice if it just lived in the list of items like you said but you've highlighted the exact reason why I didn't go that route haha (especially since it doesn't mesh very well with useAsyncList). In addition, having it statically added after your dynamic list ensures that it always exists in the collection so the loading sentinel is always rendered